app/src/lib/conviction.js

```diff
@@ -180,7 +189,22 @@ export function getRemainingTimeToPass(threshold, conviction, amount, alpha) {
   const y = threshold
   const y0 = conviction
   const x = amount
-  return Math.log(((a - 1) * y + x) / ((a - 1) * y0 + x)) / Math.log(a)
+
+  return (
+    Math.log(
+      a
+        .minus(oneBN)
+        .multipliedBy(y)
+        .plus(x)
+        .div(
+          a
+            .minus(oneBN)
+            .multipliedBy(y0)
+            .plus(x)
+        )
+        .toNumber()
+    ) / Math.log(a.toNumber())
```

Comment:
Here the `toNumber` could be removed since it will be converted automatically by the `Math.log`

Reply:
do you see any problem about making sure that we are with the right data type?

Reply:
Not really, just like to avoid redundancy, but i agree in this case makes sense to be sure with the data types.